ocaml/xapi/certificates.ml

```diff
@@ -282,12 +285,33 @@ let cert_perms kind =
   debug "%d %d" perm stat.Unix.st_perm ;
   perm
 
+let assert_valid kind ~name ~cert =
+  let cert =
+    match X509.Certificate.decode_pem (Cstruct.of_string cert) with
+    | Error (`Msg msg) ->
+        D.error "assert_valid: invalid cert. name='%s', msg='%s'" name msg ;
+        raise_corrupt kind name
+    | Ok cert ->
+        cert
+  in
+  let from', to' = X509.Certificate.validity cert in
```

Comment:
You could use `starting`,`ending` to avoid using reserved words. I think the meaning would be a little more obvious.